tests/playwright/src/model/pages/main-page.ts

```diff
@@ -94,4 +94,20 @@ export abstract class MainPage extends BasePage {
     }
     return undefined;
   }
+
+  async countRowsFromTable(): Promise<number> {
+    if (await this.pageIsEmpty()) {
+      return 0;
+    }
+    let numberOfRows = 0;
+    const table = await this.getTable();
+    const rows = await table.getByRole('row').all();
+    for (let i = rows.length - 1; i >= 0; i--) {
+      const nameCell = await rows[i].getByRole('cell').nth(3).count();
+      if (nameCell) {
+        numberOfRows++;
+      }
+    }
+    return numberOfRows;
```

Comment:
I'm not sure I understand, doesn't the all() method get all the current rows? shouldn't the value of that represent the count? Why do you need to go through each row afterwards?